### PR TITLE
Week_two.md

### PR DESCRIPTION
**Pull request by epintozzi**

### CODE REVIEW

File: module-2/backend/week_one.md [hunk 1]
```diff
@@ -3,30 +3,106 @@
...
+<br/>
+<br/>
 4. What is MVC?
+<br/>
+MVC stands for Model-View-Controller as a structure for creating a web app where each is responsible for a piece of an interconnected web app.
+<br/>
+<br/>
 5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+<br/>
+To be consistent with other applications and what other developers are doing.
```
Comment (case-eee) [inline module-2/backend/week_one.md]: what are some other reasons?

File: module-2/backend/week_one.md [hunk 2]
```diff
@@ -3,30 +3,106 @@
...
+<% ruby %> : this version executes code only
+<%= ruby %> : this version executes code and returns a result
+<br/><br/>
 15. What's an ORM?
+<br/>
+Object Relational Mapping
+<br/><br/>
 16. What's the most commonly used ORM?
+<br/>
+java?
```
Comment (case-eee) [inline module-2/backend/week_one.md]: ActiveRecord is the most commonly used Ruby ORM

File: module-2/backend/week_one.md [hunk 3]
```diff
@@ -3,30 +3,106 @@
...
+<br/>
+A migration gives us the setup to create a new table in a database
+<br/><br/>
 19. When you create a migration, does it automatically modify your database?
+<br/>
+The database is updated after we create the table and run `db:migrate`
+<br/><br/>
 20. How does a model relate to a database?
+<br/>
+A model creates the table in a database and also manages validations and relationships with other models.
```
Comment (case-eee) [inline module-2/backend/week_one.md]: a model does not actually "create" the table in the database. a model connects to a database, but it doesn't have any part of the creation.

**Comment by case-eee** [comment]
@epintozzi i made a few comments on your answers, let me know if you have any questions about anything 👍 keep up the hard work!

File: module-2/backend/week_two.md [hunk 4]
```diff
@@ -5,37 +5,201 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+
 3. How does a table relate to a model?
+</br>
+Every model has a table in the database. The table defines the attributes of the model. For example, if the model is for `student` the table might define attributes like `id`, and `name`.
+</br>
+</br>
+
 4. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.
+</br>
+They are class methods. They are used to relate tables to each other. If you have a table for `students` and one for `teachers`, a `teacher` has many students assigned to their table (probably via a `student_id`) and a `student` will belong to only one teacher (in this example). If students actually do have more than one teacher, you'd need a joiner table to relate `students` and `teachers` to each other.
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: Tricky question, but these are actually instance methods. The important thing to remember here is that you have to have a specific instance before you can call them. For example:

```ruby
jacob = Student.find(1) # gives us a specific student with an id of 1
mr_fritz = jacob.teacher # uses the `belongs_to :teacher` method on our `student` model to find Jacob's teacher (Jacob being a specific Student instance).
```

File: module-2/backend/week_two.md [hunk 5]
```diff
@@ -5,37 +5,201 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+</br>
+
+ * New instantiates a new object on a table, but does not save it unless told to do so.
+ * Create both instantiates and saves a new object on a table.
+</br>
+</br>
+
 8. At a basic level, what does cURL allow you to do?
+</br>
+cURL allows you to make GET and POST requests from the command line
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: 👍

File: module-2/backend/week_two.md [hunk 6]
```diff
@@ -5,37 +5,201 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+* Headers - This contains a lot of meta data with things like a timestamp, IP address, and format the body will be in.
+* Status Code - the code is a number between 100-600. Each number means something about whether the request was successful or not, and if not, gives a clue as to what went wrong.
+* Body - This is the information sent back; this may be empty or it may include content
+
+</br>
+</br>
+
 13. `Rack::Test` allows us to test our controllers in isolation. What are some of the methods it gives us to simulate the request/response cycle?
+</br>
+
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: This is on us. We didn't teach this time around, and should have deleted the question.

File: module-2/backend/week_two.md [hunk 7]
```diff
@@ -5,37 +5,201 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+</br>
+</br>
+
 14. Describe some techniques to make our Sinatra views more DRY. Give an example of when you would use these techniques.
+</br>
+
+* Views should not have any logic in them. Any logic that might be used over and over could be extracted out to its own method either in the model or a helper depending on what it does. One example might be when you are trying to format output to appear a certain way. You could write a method that does the formatting and call the formatted result in your view
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: 👍

File: module-2/backend/week_two.md [hunk 8]
```diff
@@ -5,37 +5,201 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+</br>
+
+* A school `has_many` teachers and a teacher `belongs_to` a school.
+
+</br>
 10. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
 11. Give an example of when you might want to store information besides ids on a join table.
+</br>
+
+* You might do this if data on the join table is distinct from any of the info stored on the tables it joins. For example on a  join table of students and teachers, there might also be an attribute indicating which classroom they are in (though there migth be an argument for separating that to its own table as well).
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: 🎉

File: module-2/backend/week_two.md [hunk 9]
```diff
@@ -5,37 +5,201 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
 14. What are some examples of acceptable values for the parts of an HTTP response?
+</br>
+
+* Header - 
+* Status code - 404
+* Body  - "Page not found"
+</br>
 15. What types of output do we want to test when we test our controllers?
+</br>
+Status codes?
```
Comment (s-espinosa) [inline module-2/backend/week_two.md]: Yup! Status codes and responses. We didn't teach this info this time around, so well done!

Comment (s-espinosa) [inline module-2/backend/week_two.md]: Yup! Status codes and responses. We didn't teach this info this time around, so well done!

Comment (s-espinosa) [inline module-2/backend/week_two.md]: Yup! Status codes and responses. We didn't teach this info this time around, so well done!

**Pull request closed without merging**